lib/internal/test_runner/reporter/spec.js

```diff
@@ -1,97 +1,35 @@
 'use strict';
-
 const {
   ArrayPrototypeJoin,
   ArrayPrototypePop,
   ArrayPrototypePush,
   ArrayPrototypeShift,
   ArrayPrototypeUnshift,
-  RegExpPrototypeSymbolSplit,
-  SafeMap,
-  StringPrototypeRepeat,
-  hardenRegExp,
 } = primordials;
 const assert = require('assert');
 const Transform = require('internal/streams/transform');
-const { inspectWithNoCustomRetry } = require('internal/errors');
 const colors = require('internal/util/colors');
 const { kSubtestsFailed } = require('internal/test_runner/test');
 const { getCoverageReport } = require('internal/test_runner/utils');
 const { relative } = require('path');
+const {
+  formatTestReport,
+  indent,
+  reporterColorMap,
+  reporterUnicodeSymbolMap,
+} = require('internal/test_runner/reporter/utils');
 
-const symbols = {
-  '__proto__': null,
-  'test:fail': '\u2716 ',
-  'test:pass': '\u2714 ',
-  'test:diagnostic': '\u2139 ',
-  'test:coverage': '\u2139 ',
-  'arrow:right': '\u25B6 ',
-  'hyphen:minus': '\uFE63 ',
-};
 class SpecReporter extends Transform {
   #stack = [];
   #reported = [];
-  #indentMemo = new SafeMap();
   #failedTests = [];
   #cwd = process.cwd();
-  #inspectOptions;
-  #colors;
 
   constructor() {
     super({ __proto__: null, writableObjectMode: true });
     colors.refresh();
-    this.#inspectOptions = { __proto__: null, colors: colors.shouldColorize(process.stdout), breakLength: Infinity };
-    this.#colors = {
```

Comment:
moving colors to utils breaks when there is no TTY, see https://ci.nodejs.org/job/node-test-commit-custom-suites-freestyle/35939/console:

`colors` cannot be deconstructed too early.

```
11:01:44         not ok 41 - test-runner/output/arbitrary-output-colored.js
11:01:44           ---
11:01:44           duration_ms: 5585.15492
11:01:44           location: '/home/iojs/build/workspace/node-test-commit-custom-suites-freestyle/test/parallel/test-runner-output.mjs:155:5'
11:01:44           failureType: 'testCodeFailure'
11:01:44           error: |-
11:01:44             Expected values to be strictly equal:
11:01:44             + actual - expected ... Lines skipped
11:01:44             
11:01:44               "{ foo: [32m'bar'[39m }\n" +
11:01:44                 '[33m1[39m\n' +
11:01:44             +   '✔ passing test [90m(*ms)[39m[39m\n' +
11:01:44             +   'ℹ tests 1[39m\n' +
11:01:44             +   'ℹ suites 0[39m\n' +
11:01:44             +   'ℹ pass 1[39m\n' +
11:01:44             +   'ℹ fail 0[39m\n' +
11:01:44             +   'ℹ cancelled 0[39m\n' +
11:01:44             +   'ℹ skipped 0[39m\n' +
11:01:44             +   'ℹ todo 0[39m\n' +
11:01:44             +   'ℹ duration_ms *[39m\n' +
11:01:44             -   '[32m✔ passing test [90m(*ms)[39m[39m\n' +
11:01:44             -   '[34mℹ tests 1[39m\n' +
11:01:44             -   '[34mℹ suites 0[39m\n' +
11:01:44             -   '[34mℹ pass 1[39m\n' +
11:01:44             -   '[34mℹ fail 0[39m\n' +
11:01:44             -   '[34mℹ cancelled 0[39m\n' +
11:01:44             -   '[34mℹ skipped 0[39m\n' +
11:01:44             -   '[34mℹ todo 0[39m\n' +
11:01:44             -   '[34mℹ duration_ms *[39m\n' +
11:01:44                 'TAP version 13\n' +
11:01:44             ...
11:01:44                 '# skipped 0\n' +
11:01:44                 '# todo 0\n' +
11:01:44                 '# duration_ms *\n'
```

Reply:
changing `reporterColorMap` to a function might be the easiest fix

Reply:
`arbitrary-output-colored` test passes on my machine. How can I reproduce the error?

Reply:
> changing `reporterColorMap` to a function might be the easiest fix

are you suggesting to replace the map with a function in utils.js or modify the constructor in spec.js?

Reply:
> test passes on my machine. How can I reproduce the error?

with `FORCE_COLOR=0 NODE_REGENERATE_SNAPSHOTS=1 ./node test/parallel/test-runner-output.mjs`

Reply:
> are you suggesting to replace the map with a function in utils.js or modify the constructor in spec.js?

either will work I think

Reply:
this might be an easier & cleaner fix: 

```diff
diff --git a/lib/internal/test_runner/reporter/utils.js b/lib/internal/test_runner/reporter/utils.js
index caa35959ee..42c4ffa3cd 100644
--- a/lib/internal/test_runner/reporter/utils.js
+++ b/lib/internal/test_runner/reporter/utils.js
@@ -28,9 +28,15 @@ const reporterUnicodeSymbolMap = {
 
 const reporterColorMap = {
   '__proto__': null,
-  'test:fail': colors.red,
-  'test:pass': colors.green,
-  'test:diagnostic': colors.blue,
+  get 'test:fail'() {
+    return colors.red;
+  },
+  get 'test:pass'() {
+    return colors.green;
+  },
+  get 'test:diagnostic'() {
+    return colors.blue;
+  },
 };
```

Reply:
> > test passes on my machine. How can I reproduce the error?
> 
> with `FORCE_COLOR=0 NODE_REGENERATE_SNAPSHOTS=1 ./node test/parallel/test-runner-output.mjs`

Still passes all tests :(

Reply:
I have made the suggested changes, please let me know if I need to add anything else.